### PR TITLE
fix: Allow users to "reactivate" their Dub subscription

### LINKED ISSUE #1331: Allow users to "reactivate" their Dub subscription
Currently, users who have canceled their Dub subscription in the past have no way to "reactivate" their subscription without creating an entirely new subscription.

This should be a super simple fix – in [this route handler](https://github.com/dubinc/dub/blob/main/apps/web/app/api/workspaces/%5BidOrSlug%5D/billing/upgrade/route.ts), we need to add an extra condition for when a user has a `stripeId` but no active `subscription`, and create the Stripe checkout session accordingly instead.

### PR DESCRIPTION
**Pull request by RajuGangitla**
Allows users with a canceled subscription to reactivate it through Stripe without needing to create a new subscription. 

Fixes #1331

### CODE REVIEW

**Comment by RajuGangitla** [comment]
@steven-tey

File: apps/web/app/api/workspaces/[idOrSlug]/billing/upgrade/route.ts [hunk 1]
```diff
@@ -16,46 +16,49 @@ export const POST = withWorkspace(async ({ req, workspace, session }) => {
...
-
-    // if the user does not have a subscription, create a new checkout session
+      });
+      return NextResponse.json({ url });
+    }
   } else {
+    // For both new users and users with canceled subscriptions
     const stripeSession = await stripe.checkout.sessions.create({
-      customer_email: session.user.email,
+      ...(workspace.stripeId ? { customer: workspace.stripeId } : { customer_email: session.user.email }),
```
Comment (TWilson023) [inline apps/web/app/api/workspaces/[idOrSlug]/billing/upgrade/route.ts]: With this change, when I attempt to upgrade a workspace that doesn't have an active subscription but does have a `stripeId`, I get the following error:
```
StripeInvalidRequestError: Tax ID collection requires updating business name on the customer. To enable tax ID collection for an existing customer, please set `customer_update[name]` to `auto`
```

**Review by TWilson023 — CHANGES_REQUESTED** [review]

File: apps/web/app/api/workspaces/[idOrSlug]/billing/upgrade/route.ts (hunk 1 above)
Comment (RajuGangitla) [reply apps/web/app/api/workspaces/[idOrSlug]/billing/upgrade/route.ts]: @TWilson023 i pushed that latest code

**Review by TWilson023 — APPROVED** [review]

**Comment by steven-tey** [comment]
@RajuGangitla thanks for the PR! Can you please sign the [CLA](https://github.com/dubinc/dub/pull/1333#issuecomment-2397442472) so we can merge the PR?

**Review by steven-tey — APPROVED** [review]

**Comment by RajuGangitla** [comment]
@steven-tey how can i sign

**Comment by steven-tey** [comment]
Thank you @RajuGangitla!

**Pull request merged**